src/material/button-toggle/button-toggle.ts

```diff
@@ -399,14 +399,12 @@ const _MatButtonToggleMixinBase: CanDisableRippleCtor & typeof MatButtonToggleBa
     '[class.mat-button-toggle-disabled]': 'disabled',
     '[class.mat-button-toggle-appearance-standard]': 'appearance === "standard"',
     'class': 'mat-button-toggle',
-    // Always reset the tabindex to -1 so it doesn't conflict with the one on the `button`,
-    // but can still receive focus from things like cdkFocusInitial.
-    '[attr.tabindex]': '-1',
```

Comment:
The `tabindex` was introduced so that the host is able to receive focus and forward it down to the internal `button` (as mentioned in the comment). I think that we still want to keep it, otherwise setting `cdkFocusInitial` on the toggle won't work anymore. Does the screen reader issue still occur if we have both the `tabindex` and the `role="presentation"`?

If there's no alternative to removing the `tabindex`, another way to resolve the issue could be to change the focus trap so that if a `cdkFocusInitial` element isn't focusable, we would try to look for a focusable element inside of it.

Reply:
The screen reader issue still happens with both `tabindex` and `role="presentation`.
I changed `focusInitialElement` in focus trap to look for a focusable child if the `redirectToElement` isn't focusable